Update changelog for 0.12.0 release

APTy opened the pull request:
Also adds the 0.11.2 changelog that was never cherry-picked

APTy commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Changed datastore versioning to be compatible with a 1.0 release (#2008)
+
+### Deprecated
+-
```
does anyone remember if anything was deprecated this release? i looked over the PRs quickly, but nothing popped out in my mind

azdagron replied:
Just a few come to mind:
- Registration and Node API are deprecated with this release (we now log on use). #1997.
- `registration_api` configuration section is deprecated in favor of `server_api` in the k8s-workload-registrar. #2001.

evan2645 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Added `ExpiresAt` to `entry show` output (#1973)
+- Added improved logging (#1996, #1997)
+- Added `k8s_psat:agent_node_ip` selector (#1979)
+- Added support for HTTP/2 to the Server (#1968)
```
@azdagron @rturner3 is this accurate?

azdagron replied:
This is a little misleading. The server already supported HTTP/2 (which is the transport that gRPC uses). We just fixed a bug where we weren't announcing our support for HTTP/2 during the TLS handshake which caused some gRPC libraries to fail to connect to the server. This should probably go under "Fixed"?

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Changed internals to rely on new server APIs (#1849, #1878, #1907, #1908, #1909, #1913, #1947, #1982, #1998, #2001)
+- Changed Workload API to return a standardized JWKS object (#1904)
+- Changed Registration API to validate selector values (#1919)
+- Changed log message casing and punctuation (#1950, #1952)
```
```suggestion
- Changed log message casing and punctuation to be consistent with project guidelines (#1950, #1952)
```

evan2645 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Added `k8s_psat:agent_node_ip` selector (#1979)
+- Added support for HTTP/2 to the Server (#1968)
+
+### Changed
```
Since we have other non-user-facing changes listed here, should we also list go version bump? https://github.com/spiffe/spire/pull/1961

Similar question for changing up our grpc dependencies #1986

APTy replied:
i've removed more of the non-user-facing changes, so will probably keep these out unless you think they should be added back

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Changed Workload API to return a standardized JWKS object (#1904)
+- Changed Registration API to validate selector values (#1919)
+- Changed log message casing and punctuation (#1950, #1952)
+- Changed datastore versioning to be compatible with a 1.0 release (#2008)
```
Does this have any user facing impact? Do we need to mention it?

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+
+### Changed
+- Changed the agent to crash when it needs to re-attest (#1797)
+- Changed internals to rely on new server APIs (#1849, #1878, #1907, #1908, #1909, #1913, #1947, #1982, #1998, #2001)
```
Does this have any user facing impact? Do we need to mention it?

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Added support for HTTP/2 to the Server (#1968)
+
+### Changed
+- Changed the agent to crash when it needs to re-attest (#1797)
```
"crash" has a pretty negative connotation. Maybe something a little softer?
```suggestion
- Changed the agent to shut down when it is no longer attested (#1797)
```

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Changed the agent to crash when it needs to re-attest (#1797)
+- Changed internals to rely on new server APIs (#1849, #1878, #1907, #1908, #1909, #1913, #1947, #1982, #1998, #2001)
+- Changed Workload API to return a standardized JWKS object (#1904)
+- Changed Registration API to validate selector values (#1919)
```
Maybe this should go under "Fixed"? Also, it isn't really validating the values (i.e. is this a selector recognizable by my configured plugins), but rather just making sure the syntax is consistent.

Maybe something like:
```
Fixed Registration API to validate selector syntax (#1919)
```

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+-
+
+### Removed
+- Removed agent FetchSVIDTTL gauge metrics and other telemtry (#1881, #1946)
```
Typo. What's the benefit of calling out FetchSVIDTTL gauge specifically? I _think_ we can just be generic here?
```suggestion
- Removed some superfluous or otherwise unusable metrics and labels (#1881, #1946,#2004)
```

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+
+### Removed
+- Removed agent FetchSVIDTTL gauge metrics and other telemtry (#1881, #1946)
+- Removed datastore communication over gRPC - the SQL datastore plugin is now loaded directly in the Server process (#1938)
```
Not sure this needs to be in the changelog, as it isn't really user facing?

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+### Removed
+- Removed agent FetchSVIDTTL gauge metrics and other telemtry (#1881, #1946)
+- Removed datastore communication over gRPC - the SQL datastore plugin is now loaded directly in the Server process (#1938)
+- Removed logging of JWT-SVIDs that fail validation (#1953)
```
I'd almost consider this a security fix?

evan2645 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+### Added
+- Added debug endpoints (#1792)
+- Added and improved documentation (#1806, #1808, #1813, #1821, #1831, #1837, #1838, #1861, #1868, #1869, #1876, #1880, #1883, #1886, #1902, #1918, #1923, #1939, #1951, #1970, #1999, #2011)
+- Added improved testing (#1837, #1872, #1873, #1931, #1937, #1942, #1948, #1954, #1958, #1978, #1982)
```
I'm not sure the merits of including this line and the next in the changelog. It's nice/reassuring to know that this work is happening, but is it important to consumers of the changelog? Particularly with the documentation line, it's hard to know exactly _what_ improved

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Removed agent FetchSVIDTTL gauge metrics and other telemtry (#1881, #1946)
+- Removed datastore communication over gRPC - the SQL datastore plugin is now loaded directly in the Server process (#1938)
+- Removed logging of JWT-SVIDs that fail validation (#1953)
+- Removed SPIFFE ID label from metrics (#2004)
```
Maybe lump this into the changelog line about metrics above?

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Added caching for registration entries and agent selectors (#1859, #1896, #1943, #1944, #1956)
+- Added database indexes to attested node columns (#1912)
+- Added better configuration for Vault UpstreamAuthority plugin (#1871, #1965, #1981)
+- Added delete mode for federated bundles from the bundle API (#1897)
```
```suggestion
- Added delete mode for federated bundles to the Bundle API (#1897)
```

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Added improved metrics handling (#1885, #1925, #1932)
+- Added caching for registration entries and agent selectors (#1859, #1896, #1943, #1944, #1956)
+- Added database indexes to attested node columns (#1912)
+- Added better configuration for Vault UpstreamAuthority plugin (#1871, #1965, #1981)
```
What exactly is better?
```suggestion
- Added support for roles, namespaces, and re-authentication to the Vault UpstreamAuthority plugin (#1871, #1965, #1981)
```

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Added better configuration for Vault UpstreamAuthority plugin (#1871, #1965, #1981)
+- Added delete mode for federated bundles from the bundle API (#1897)
+- Added reading JSON from STDIN for CLI entry create/update commands (#1905)
+- Added better middleware for Agent APIs (#1936)
```
No user impact. Maybe we should leave this out?

azdagron commented on CHANGELOG.md:
```diff
@@ -1,5 +1,55 @@
+- Added better middleware for Agent APIs (#1936)
+- Added support for multiple CA bundle files in x509pop (#1949)
+- Added `ExpiresAt` to `entry show` output (#1973)
+- Added improved logging (#1996, #1997)
```
Maybe we can lump this in with the "improved telemetry line"?

evan2645 reviewed (dismissed):
🚀 thank you for this @APTy! shaped up to be a nice release

I left a bunch of feedback/thoughts/questions, none of which I think are really blocking. The new changelog format is great, but I think we need to figure out where we draw the line w.r.t. how detailed we get in publicizing non-user-facing changes

azdagron reviewed (commented):
Thanks for combing through the huge list of changes and putting this together, @APTy !

I hope my comments aren't too nitpicky.

APTy commented:
@azdagron @evan2645 I removed much of the non-user-facing changes, and changed some of the verbiage (no more `Added` prefix to every line, which doesn't seem entirely necessary). Let me know if you have any final thoughts!

azdagron reviewed (approved):
\o/

amartinezfayo reviewed (commented):
Are you adding the version number and date?
Looks good to me otherwise.

amartinezfayo reviewed (approved):
🎉
